lib/dry/transaction/step.rb

```diff
@@ -29,8 +29,7 @@ def initialize(step_adapter, step_name, operation_name, operation, options, call
         @call_args = call_args
       end
 
-      # TODO: rename to curry
-      def apply(operation: UNDEFINED, call_args: UNDEFINED)
+      def curry(operation: UNDEFINED, call_args: UNDEFINED)
```

Comment:
I think my TODO here was actually out of date. I don't think curry as a name makes sense if we're passing more than just call args. Would you mind renaming again to `#with`? That's more consistent with similar methods elsewhere in dry-rb land.